src/main/java/com/google/sps/servlets/LoadInterviewsServlet.java

```diff
@@ -0,0 +1,256 @@
+    List<ArrayList<PossibleInterviewSlot>> possibleInterviewsForWeek =
+        new ArrayList<ArrayList<PossibleInterviewSlot>>();
+
+    if (!possibleInterviews.isEmpty()) {
+      ArrayList<PossibleInterviewSlot> dayOfSlots = new ArrayList<PossibleInterviewSlot>();
+      for (PossibleInterviewSlot possibleInterview : possibleInterviews) {
+        if (!possibleInterview.date().equals(date)) {
+          possibleInterviewsForWeek.add(dayOfSlots);
+          dayOfSlots = new ArrayList<PossibleInterviewSlot>();
+          date = possibleInterview.date();
+        }
+        dayOfSlots.add(possibleInterview);
+      }
+      possibleInterviewsForWeek.add(dayOfSlots);
+    }
+
+    request.setAttribute("weekList", possibleInterviewsForWeek);
+    RequestDispatcher rd = request.getRequestDispatcher("/possibleInterviewTimes.jsp");
+
+    try {
+      rd.forward(request, response);
+    } catch (ServletException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  // Uses an Instant and a timezoneOffset to create a ZonedDateTime instance.
+  private static ZonedDateTime generateDay(Instant instant, ZoneOffset timezoneOffset) {
+    return instant.atZone(ZoneId.ofOffset("UTC", timezoneOffset));
+  }
+
+  // This method takes the timezoneOffsetMinutes int and converts it
+  // into a proper ZoneOffset instance.
+  private static ZoneOffset convertIntToOffset(int timezoneOffsetMinutes) {
+    return ZoneOffset.ofHoursMinutes((timezoneOffsetMinutes / 60), (timezoneOffsetMinutes % 60));
+  }
+
+  private List<PossibleInterviewSlot> getPossibleInterviewSlots(
+      TimeRange range, ZoneOffset timezoneOffset) {
+
+    Set<String> interviewers = availabilityDao.getUsersAvailableInRange(range.start(), range.end());
+    Set<PossibleInterviewSlot> possibleInterviews = new HashSet<PossibleInterviewSlot>();
+
+    // We don't want to schedule an interview for a user with themself, so we are removing
+    // the current user's id from the list.
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail();
+    String userId = userService.getCurrentUser().getUserId();
+
+    // Since UserId does not have a valid Mock, if the id is null (as when testing), it will be
+    // replaced with this hashcode.
+    if (userId == null) {
+      userId = String.format("%d", userEmail.hashCode());
+    }
+
+    interviewers.remove(userId);
+
+    for (String interviewer : interviewers) {
+      possibleInterviews.addAll(
+          getPossibleInterviewSlotsForPerson(interviewer, range, timezoneOffset));
+    }
+
+    // We need to check that the person looking to schedule is not already scheduled during any of
+    // the proposed times.
+
+    List<ScheduledInterview> userScheduledInterviews =
+        scheduledInterviewDao.getScheduledInterviewsInRangeForUser(
+            userId, range.start(), range.end());
+
```

Comment:
Nit: drop most of these blank lines (up until L172, I think).

The blank lines are intended to help readers understand logical chunks.

Reply:
Done